web/src/hotkey.js

```diff
@@ -798,6 +798,46 @@ export function process_hotkey(e, hotkey) {
         return true;
     }
 
+    if (event_name === "toggle_compose_preview" && $(".message_edit").is(":visible") && !$(".new_message_textarea").is(":focus")) {
+        if ($(".compose-control-buttons-container .markdown_preview").is(":visible")) {
+            e.preventDefault();
+            const $row = $(".message_edit_form");
+            const $msg_edit_content = $row.find(".message_edit_content");
+            const content = $msg_edit_content.val();
+
+            // Disable unneeded compose_control_buttons as we don't
+            // need them in preview mode.
+            $row.addClass("preview_mode");
+            $row.find(".preview_mode_disabled .compose_control_button").attr("tabindex", -1);
+
+            $msg_edit_content.hide();
+            $row.find(".markdown_preview").hide();
+            $row.find(".undo_markdown_preview").show();
+            $row.find(".preview_message_area").show();
+
+            compose.render_and_show_preview(
+                $row.find(".markdown_preview_spinner"),
+                $row.find(".preview_content"),
+                content,
+            );
+        } else {
+            e.preventDefault();
+            const $row = $(".message_edit_form");
+
+            // While in preview mode we disable unneeded compose_control_buttons,
+            // so here we are re-enabling those compose_control_buttons
+            $row.removeClass("preview_mode");
+            $row.find(".preview_mode_disabled .compose_control_button").attr("tabindex", 0);
+
+            $row.find(".message_edit_content").show();
+            $row.find(".undo_markdown_preview").hide();
+            $row.find(".preview_message_area").hide();
+            $row.find(".preview_content").empty();
+            $row.find(".markdown_preview").show();
+        }
+        return true;
+    }
```

Comment:
This should definitely be calling a function like `compose.show_preview_area`, not doing a lot of custom code here.

Reply:
compose.show_preview_area adds preview only to the div with id compose, and while in edit mode the parent div doesn't have the id compose. that's why in the code I'm selecting the div with class message_edit_form and adding preview_mode to it.

![image](https://github.com/user-attachments/assets/f91db0fe-ee66-4106-8004-7f60772907eb)

Reply:
I have created new function for show and hide preview in edit mode, and used them instead of custom code directly inside If statements.